pkg/policy/resolve.go

```diff
@@ -86,6 +102,85 @@ func getSecurityIdentities(labelsMap cache.IdentityCache, selector *api.Endpoint
 	return identities
 }
 
+// DistillPolicy filters down the specified IdentityPolicy (which acts upon
+// selectors) into a set of concrete map entries based on the specified
+// identityCache. These can subsequently be plumbed into the datapath.
+//
+// Must be performed while holding the Repository lock.
+func (p *IdentityPolicy) DistillPolicy(policyOwner PolicyOwner, identityCache cache.IdentityCache) *EndpointPolicy {
+
+	calculatedPolicy := &EndpointPolicy{
+		IdentityPolicy:          p,
+		PolicyMapState:          make(MapState),
+		PolicyOwner:             policyOwner,
+		DeniedIngressIdentities: cache.IdentityCache{},
+		DeniedEgressIdentities:  cache.IdentityCache{},
+	}
+
+	labels := policyOwner.GetSecurityIdentity().LabelArray
```

Comment:
There is a possibility that the returned value from `GetSecurityIdentity()` could be nil since it returns a pointer.

Reply:
Good spotting. This actually highlights that this function shouldn't receive a `PolicyOwner` (which represents an endpoint), but instead it should receive the identity directly. Changing this over should tidy up the interface quite nicely.

Reply:
On closer inspection, we're still depending on endpoint-specific state here for the moment so can't switch it over. Will be able to tidy this up when #6921 lands.